api/interactives.go

```diff
@@ -212,15 +205,23 @@ func (api *API) UpdateInteractiveHandler(w http.ResponseWriter, req *http.Reques
 		}
 	}
 
-	// 5. write to DB
+	// 6. write to DB
 	err = api.mongoDB.UpsertInteractive(ctx, id, updatedModel)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		log.Error(ctx, "Unable to write to DB", err)
 		return
 	}
 
-	WriteJSONBody(updatedModel, w, http.StatusOK)
+	// 7. get updated model
```

Comment:
i think this is important - if we dont update the model in request in expected way for whatever reason we will send back an incorrect model to whats in db. its a shame we cant update and get document in one request :( 

the update might not be the full interactive too - so archive might not come in every update request.